### PR TITLE
Added test implementation

### PR DESCRIPTION
**Pull request by g-ortuno**
Please review @schilit

### CODE REVIEW

File: android-uribeacon/uribeacon-library/src/main/java/org/uribeacon/beacon/UriBeacon.java [hunk 1]
```diff
@@ -484,5 +499,33 @@ private static int totalBytes(String uriString) {
...
+    int currentPos = 0;
+    try {
+      while (currentPos < scanRecord.length) {
+        int fieldLength = scanRecord[currentPos++] & 0xff;
+        if (fieldLength == 0) {
+          break;
+        }
+        int fieldType = scanRecord[currentPos++] & 0xff;
+        if (fieldType == DATA_TYPE_SERVICE_DATA) {
+          if (scanRecord[currentPos++] == TEST_SERVICE_16_BIT_UUID_BYTES[0]
```
Comment (schilit) [inline android-uribeacon/uribeacon-library/src/main/java/org/uribeacon/beacon/UriBeacon.java]: I know you just copied the pattern but do you see the bug here? You can increment currentPos, fall out of the block and then increment currentPso += fieldLength -1; These increments should be offset: currentPos + 2; etc.

Comment (g-ortuno) [inline android-uribeacon/uribeacon-library/src/main/java/org/uribeacon/beacon/UriBeacon.java]: I see it. Isn't the bug also present in the original code then?

Comment (schilit) [inline android-uribeacon/uribeacon-library/src/main/java/org/uribeacon/beacon/UriBeacon.java]: yes, that is why I said "I know you just copied the pattern"

**Comment by g-ortuno** [comment]
Ready for review again @schilit

**Pull request merged**